qmerge.c

```diff
@@ -960,9 +994,17 @@ pkg_merge(int level, const depend_atom *atom, const struct pkg_t *pkg)
 
 	/* extract the binary package data */
 	mkdir("image", 0755);
-	snprintf(buf, sizeof(buf),
-			BUSYBOX " tar -%sx%sf %s.tar.bz2 -C image/",
-			compr, ((verbose > 1) ? "v" : ""), pkg->PF);
+	if (compr[0] != 'I') {
+		snprintf(buf, sizeof(buf),
+			BUSYBOX " tar -x%s%s -f %s.tar.bz2 -C image/",
+			((verbose > 1) ? "v" : ""), compr, pkg->PF);
+	} else {
+		/* tar -I ... does not work with busybox, so we do pipe. */
+		/* Redirect stdin to avoid file extension test of e.g. zstd. */
+		snprintf(buf, sizeof(buf),
+			BUSYBOX " %s -dc <%s.tar.bz2 | tar -x%sf - -C image/",
+			compr + 2, pkg->PF, ((verbose > 1) ? "v" : ""));
```

Comment:
the main meat of this change:

```
% busybox lzip
lzip: applet not found
% busybox zstd
zstd: applet not found
```
I see no ways to enable these in busybox, so is this ever going to work? Or is this some optimistic approach for when busybox is not in effect and people have the appropriate (de)compressors installed?

Reply:
One cannot expect that a zstd-compressed file will be installed if there is no zstd binary in path.

It was not clear to me that BUSYBOX would just extend to "busybox". I would have expected that it would expand to "busybox sh -c" or something like that so that the remainder of the line is actually interpreted as in a "system" call by busybox.
If my this assumptions was false, I see 2 possibilities to fix the "tar -I" case:

1. Use `BUSYBOX " sh -c '%s -d <%s.tar.bz2 | tar -x%sf - -C image'"`
2. Omit BUSYBOX and call `xsystem("tar -I ...")`, relying that tar has the -I option

I suppose that 1. is more compatible.

Reply:
busybox is used on embedded systems, where gnu tar isn't available, hence the awkward workaround

I think we should just document (in the comments) that we expect this to fail for busybox, but will work fine for people who have the necesary stuff installed.  We may want to show an appropriate error that highlights the exact problem (binpkg compressed using X, decompressor not found)